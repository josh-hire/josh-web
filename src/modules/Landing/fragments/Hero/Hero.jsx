import { Title, Text, Container, Image, Flex, Col, Grid, Button } from '@mantine/core';
import styles from './styles.module.scss';

export function HeroSection() {
  return (
    <Container className={styles.root}>
      <Grid columns={12}>
        <Col span={6}>
          <Flex direction={'column'} h={'100%'} justify={'center'} w={'100%'}>
            <Title className={styles.title}>Caffeine Composers</Title>
            <Title className={styles.title}>get swiping</Title>
            <Text className={styles.description}>
              {`Getahead has launched the world's fastest jobs app`}
            </Text>

            <Button className={styles.button}>
              Download the app
            </Button>
          </Flex>
        </Col>
        <Col span={6}>
          <div className={styles.heroImage}>
            <Image alt="Logo PT_tester" src={'/assets/home/images/hero.png'} />
          </div>
        </Col>
      </Grid>
    </Container>
  );
}
