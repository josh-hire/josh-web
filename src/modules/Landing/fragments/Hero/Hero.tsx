import { Title, Text, Container, Image, Flex, Col, Grid, Button } from '@mantine/core';
import styles from './styles.module.scss';
import MotionAnimate from 'Components/elements/MotionAnimate';
import Link from 'next/link';

export function HeroSection() {
  return (
    <MotionAnimate
      animate="fadeIn"
      autoAnimate={true}
      key={'heroMotion'}
      style={{
        animationDuration: '1.5s',
      }}>
      <Container className={styles.root} size={'lg'}>
        <Grid columns={12}>
          <Col span={12}>
            <Title className={styles.title}>SWIPE through jobs and get HIRED fast</Title>
          </Col>
        </Grid>
        <Grid columns={12}>
          <Col span={6}>
            <div className={styles.heroImage}>
              <Image alt="Logo PT_tester" src={'/assets/home/images/hero.png'} />
            </div>
          </Col>
          <Col span={6}>
            <Flex direction={'column'} h={'100%'} justify={'end'} align={'end'} w={'100%'}>
              <Title className={styles.title}>Swipe to Hire</Title>
              <Text className={styles.description}>
                {`Unlock your full potential! The journey to greatness begins with one swipeâ€”explore endless opportunities, connect with your dream job, and take the first step toward a future filled with success.`}
              </Text>
              <Flex direction={'row'} justify={'end'} align={'end'} w={'100%'} mt={20}>
                <Button className={styles.button}>Sign In</Button>
                <Link href={'/register'}>
                  <Button className={styles.button}>Sign Up</Button>
                </Link>
              </Flex>
            </Flex>
          </Col>
        </Grid>
      </Container>
    </MotionAnimate>
  );
}
