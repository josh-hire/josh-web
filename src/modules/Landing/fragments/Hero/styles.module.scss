.root{
  display: flex;
  background-color: black;
  height: auto;
  padding-top: 10rem;
  padding-bottom: 8rem;
  width: 100%;

}

.title{
  font-size: 45px;
  color: white;
}
.description{
  font-size:18px;
  font-weight: 600;
  text-align:left;
  color: #7e7e7e;
  margin-top: 1.5rem;
  margin-bottom: 1.5rem;
}

.heroImage{
  z-index: 999;
  position: relative;
  width: auto;
  height: auto;
  right: -6rem;
}

.button {
  align-items: center;
  background-color: #fd91f5;
  border-radius: 24px;
  display: flex;
  width: 161px;
  flex-direction: row;
  gap: 8px;
  height: 48px;
  justify-content: center;
  padding: 10px 12px;
  color: white;
  cursor: pointer;
  margin-right: 16px;

  &:hover {
    background-color: rgba(255, 255, 255, 0.2);
    color: white;
    border: 1px solid white;
    transition: 0.1s all ease-in-out;
  }
}