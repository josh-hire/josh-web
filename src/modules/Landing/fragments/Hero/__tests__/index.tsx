import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { HeroSection } from '../Hero';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      hl: 'id',
    },
  }),
}));

describe('src/modules/Landing/fragments/Hero', () => {
  test('should be able to render Hero', () => {
    const root = render(<HeroSection />);
    expect(root).toMatchSnapshot();
  });
});
