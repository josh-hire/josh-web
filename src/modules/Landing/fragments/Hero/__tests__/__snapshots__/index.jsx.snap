// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/modules/Landing/fragments/Hero should be able to render Hero 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mantine-Container-root root mantine-15pjuqq"
      >
        <div
          class="mantine-Grid-root mantine-nsdj"
        >
          <div
            class="mantine-Grid-col mantine-fhjbsl"
          >
            <div
              class="mantine-129nl33"
            >
              <h1
                class="mantine-Text-root mantine-Title-root title mantine-dyfpg1"
              >
                Caffeine Composers
              </h1>
              <h1
                class="mantine-Text-root mantine-Title-root title mantine-dyfpg1"
              >
                get swiping
              </h1>
              <div
                class="mantine-Text-root description mantine-1pbxw0k"
              >
                Getahead has launched the world's fastest jobs app
              </div>
              <button
                class="mantine-UnstyledButton-root mantine-Button-root button mantine-8nr514"
                data-button="true"
                type="button"
              >
                <div
                  class="mantine-3xbgk5 mantine-Button-inner"
                >
                  <span
                    class="mantine-qo1k2 mantine-Button-label"
                  >
                    Download the app
                  </span>
                </div>
              </button>
            </div>
          </div>
          <div
            class="mantine-Grid-col mantine-fhjbsl"
          >
            <div
              class="heroImage"
            >
              <div
                class="mantine-Image-root mantine-15po0m8"
                style="width: 100%;"
              >
                <figure
                  class="mantine-11nhzn5 mantine-Image-figure"
                >
                  <div
                    class="mantine-qqmv3w mantine-Image-imageWrapper"
                  >
                    <img
                      alt="Logo PT_tester"
                      class="mantine-3y8yz3 mantine-Image-image"
                      src="/assets/home/images/hero.png"
                      style="object-fit: cover; width: 100%; height: auto;"
                    />
                  </div>
                </figure>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mantine-Container-root root mantine-15pjuqq"
    >
      <div
        class="mantine-Grid-root mantine-nsdj"
      >
        <div
          class="mantine-Grid-col mantine-fhjbsl"
        >
          <div
            class="mantine-129nl33"
          >
            <h1
              class="mantine-Text-root mantine-Title-root title mantine-dyfpg1"
            >
              Caffeine Composers
            </h1>
            <h1
              class="mantine-Text-root mantine-Title-root title mantine-dyfpg1"
            >
              get swiping
            </h1>
            <div
              class="mantine-Text-root description mantine-1pbxw0k"
            >
              Getahead has launched the world's fastest jobs app
            </div>
            <button
              class="mantine-UnstyledButton-root mantine-Button-root button mantine-8nr514"
              data-button="true"
              type="button"
            >
              <div
                class="mantine-3xbgk5 mantine-Button-inner"
              >
                <span
                  class="mantine-qo1k2 mantine-Button-label"
                >
                  Download the app
                </span>
              </div>
            </button>
          </div>
        </div>
        <div
          class="mantine-Grid-col mantine-fhjbsl"
        >
          <div
            class="heroImage"
          >
            <div
              class="mantine-Image-root mantine-15po0m8"
              style="width: 100%;"
            >
              <figure
                class="mantine-11nhzn5 mantine-Image-figure"
              >
                <div
                  class="mantine-qqmv3w mantine-Image-imageWrapper"
                >
                  <img
                    alt="Logo PT_tester"
                    class="mantine-3y8yz3 mantine-Image-image"
                    src="/assets/home/images/hero.png"
                    style="object-fit: cover; width: 100%; height: auto;"
                  />
                </div>
              </figure>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
